src/main/java/sparksoniq/jsoniq/runtime/iterator/functions/DynamicFunctionCallIterator.java

```diff
@@ -73,82 +74,10 @@ public DynamicFunctionCallIterator(
     @Override
     public void openLocal() {
         setFunctionItemAndIteratorWithCurrentContext();
```

Comment:
@CanBerker does this apply here too that it is redundant?

Reply:
Precisely!

Reply:
Very good. For dynamic calls recursion is not an issue as I do not think it is (as of the current version of the W3C spec) possible for a function item to call itself.

Reply:
I reverted, it seems that it does make a difference. We could investigate later on.

Reply:
I think I found the issue. It is because there was this partial application test, but it is redundant with the other iterator. Now it should work.

Reply:
I put it back for now -- there seems to be an issue with serializing and deserializing function items that forces us to reset the function when opening, which we can investigate later.